planner/core/physical_plans.go

```diff
@@ -404,6 +406,20 @@ type PhysicalHashAgg struct {
 	basePhysicalAgg
 }
 
+// NewPhysicalHashAgg creates a new PhysicalHashAgg from a LogicalAggregation.
+func NewPhysicalHashAgg(la *LogicalAggregation, prop *property.PhysicalProperty, taskTp ...property.TaskType) *PhysicalHashAgg {
```

Comment:
Why we need to `taskTp` since there is already `prop`?

Reply:
This function is also used in planner/core, for `exhausPhysicalPlans`. It may generate `CopSingleReadTaskType`, `CopDoubleReadTaskType` and `RootTaskType`. I just extract this function to be reused in cascades package.

Reply:
But since it can only return one `HashAgg`, why do we need to handle multiply task types? The `taskTp` should be already filled in `prop`.

Reply:
I use `taskTp ...property.TaskType` because I want to call this function by both `NewPhysicalHashAgg(la, oldProp)` and `NewPhysicalHashAgg(la, oldProp, taskTp)`. 
If it is confusing, How about this:
```golang
// NewPhysicalHashAgg creates a new PhysicalHashAgg from a LogicalAggregation.
func NewPhysicalHashAgg(la *LogicalAggregation, expectedCnt float64, prop *property.PhysicalProperty) *PhysicalHashAgg {
	agg := basePhysicalAgg{
		GroupByItems: la.GroupByItems,
		AggFuncs:     la.AggFuncs,
	}.initForHash(la.ctx, la.stats.ScaleByExpectCnt(expectedCnt), la.blockOffset, prop)
	agg.SetSchema(la.schema.Clone())
	return agg
}
```
and called from `getHashAggs` in `exhaust_physical_plans.go`:
```golang
	for _, taskTp := range taskTypes {
		agg := NewPhysicalHashAgg(la, prop.ExpectedCnt, &property.PhysicalProperty{ExpectedCnt: math.MaxFloat64, TaskTp: taskTp})
		hashAggs = append(hashAggs, agg)
	}
```

Reply:
Ok, I think it is better now.